lib/board.ex

```diff
@@ -0,0 +1,41 @@
+defmodule Board do
+  defstruct [:board]
+
+  def new do
+    %Board{board: empty_board()}
+  end
+
+  defp empty_board do
+    %{
+      0 => empty_row(),
+      1 => empty_row(),
+      2 => empty_row(),
+      3 => empty_row(),
+      4 => empty_row(),
+      5 => empty_row(),
+      6 => empty_row(),
+      7 => empty_row(),
+      8 => empty_row(),
+      9 => empty_row()
+    }
+  end
+
+  defp empty_row do
+    %{
+      0 => empty_tile(),
+      1 => empty_tile(),
+      2 => empty_tile(),
+      3 => empty_tile(),
+      4 => empty_tile(),
+      5 => empty_tile(),
+      6 => empty_tile(),
+      7 => empty_tile(),
+      8 => empty_tile(),
+      9 => empty_tile()
+    }
+  end
+
+  defp empty_tile do
+    %{selected: false}
```

Comment:
Have you ever read about [state machines](https://en.wikipedia.org/wiki/Finite-state_machine)? If not, they are a super useful concept we should explore some time.

In my opinion, using Elixir atoms and Ruby keywords can make your code more readable and meaningful when you're representing different states or simple booleans like this. For example, representing an empty tile with the atom `:empty` instead of a map of properties. You'd write conditions like `tile == :empty` instead of `tile.selected == false`.

If a tile is a bucket of properties and we derive its state by checking them in combination, there's a possible future that grows pretty complex pretty quick—`%{selected: false}` grows to `%{selected: false, has_bomb: false, has_flag: true}`, etc. It's easy to introduce weird states that shouldn't be possible as the number of these properties grows.

You might need to reach for a more complex representation eventually anyways, like when you need to represent "has a bomb and is flagged." (If you start creating states with "and" in the name, like `:has_bomb_and_is_flagged`, that's a clue). In either event, the same advice as above applies: try and hide all this behind an interface you own.

Reply:
I've taken your suggestion and replaced `%{selected: false}` with `:empty` in a later branch. Thank you! :)